app/src/main/java/org/mozilla/focus/experiments/ExperimentalFeatures.kt

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.experiments
+
+import mozilla.components.service.nimbus.NimbusApi
+
+private const val EXPERIMENT_SINGLE_TAB_BROWSER = "single-tab-browser"
+private const val BRANCH_TREATMENT = "treatment"
+
+/**
+ * Feature flags driven by Nimbus experiments.
+ */
+class ExperimentalFeatures(nimbus: NimbusApi) {
+    /**
+     * Whether or not this browser should have multiple tabs or run in single-tab mode.
+     */
+    val hasMultipleTabs: Boolean by lazy {
```

Comment:
I am curious about the impact of using `by lazy` here: I want to check only once per app launch for this experiment. Because it would be confusing for the user if they have multiple tabs and then we switch to single-tab mode. So while the app is running this should not change.
But I guess the experiment configuration in Nimbus could change while the app is running. Would this cause us to report the wrong data (e.g. "User is in treatment" but actually this feature is not enabled because they were not when we checked?)

Reply:
The way Nimbus is meant to be used, the experiment treatment should _never_ change during the execution of the app. The only API function that can cause this to happen is `Nimbus.applyPendingExperiments` and only if there are experiments that need applied. This function should be set up to only be called when the app is launched so that Nimbus will only return the same value for a feature configuration during that execution. Nimbus even caches the value in memory so that retrieving it shouldn't trigger a read from storage.

Reply:
Great, thank you. That sounds like exactly the behavior I need. And since we only call `applyPendingExperiments` when constructing `NimbusApi` this seems to be safe: https://github.com/mozilla-mobile/focus-android/blob/5a990f5d90110be0d24710e216e82139a87430a2/app/src/main/java/org/mozilla/focus/experiments/NimbusSetup.kt#L81